# A base editing platform for the correction of cancer driver mutations unmasks conserved p53 transcription programs

This repository contains the accompanying script used for the differential expression profiling of the RNA-Seq data.

The raw data from the three cell lines were aligned to the human genome hg38 using STAR aligner v2.7.3a after quality check with FastQC v0.12.1. The annotation used for mapping was GENCODE v46. Consequently, using featureCounts v2.0.6 a count matrix was generated for 42 samples. The code for the DE analysis has been included here. All genes with no reads were discarded. Principal component analysis (PCA) was performed using genes with the highest variance. The differential expression analysis was performed using DESeq2. The fold change cutoff was set at 2 and p-value threshold was set at 0.05 for considering the differentially expressed genes. Heatmaps were clustered based on hierarchical clustering. Gene ontology analysis was performed with the goseq algorithm.
